scripts/patches/tor-reload.diff

```diff
@@ -22,27 +12,40 @@ index 123c33a..cdf12ba 100644
        //log_debug(LD_NET, "Already have %s on %s:%d",
        //    conn_type_to_string(found_port->type), conn->address, conn->port);
 diff --git a/src/or/main.c b/src/or/main.c
-index bd4f7ea..0037e6b 100644
+index 66a8571..389f753 100644
 --- a/src/or/main.c
 +++ b/src/or/main.c
-@@ -1922,6 +1922,16 @@ retry_listeners_callback(time_t now, const or_options_t *options)
- }
+@@ -96,6 +96,7 @@ static void connection_start_reading_from_linked_conn(connection_t *conn);
+ static int connection_should_read_from_linked_conn(connection_t *conn);
+ static int run_main_loop_until_done(void);
+ static void process_signal(int sig);
++static void tor_reload_inner(void);
  
- static int
-+force_retry_listeners_callback(time_t now, const or_options_t *options)
-+{
-+  (void)now;
-+  (void)options;
-+  retry_all_listeners(NULL, NULL, 1);
-+  retry_all_listeners(NULL, NULL, 0);
-+  return PERIODIC_EVENT_NO_UPDATE;
-+}
+ /********* START VARIABLES **********/
+ int global_read_bucket; /**< Max number of bytes I can read this second. */
+@@ -170,6 +171,9 @@ static int can_complete_circuits = 0;
+  */
+ int quiet_level = 0;
+ 
++/** Flag: Set to true if you want to reload tor */
++static int should_reload = 0;
 +
-+static int
- expire_old_ciruits_serverside_callback(time_t now, const or_options_t *options)
- {
-   (void)options;
-@@ -3192,7 +3202,7 @@ tor_cleanup(void)
+ /********* END VARIABLES ************/
+ 
+ /****************************************************************************
+@@ -2368,6 +2372,11 @@ run_main_loop_once(void)
+ 
+   if (nt_service_is_stopping())
+     return 0;
++    
++  if (should_reload == 1) {
```

Comment:
To make it 100% bulletproof I'd protect it (the whole reload) with a mutex, something like:
`tor_mutex_acquire(&reload_lock); ... tor_mutex_release(&reload_lock);`

...where `reload_lock` is declared as static mutext in the lines of: 

`static tor_mutex_t *reload_lock = NULL;`